HealthcareSystem/UserService/Model/PatientAccount.cs

```diff
@@ -1,35 +1,45 @@
-﻿using System;
-using System.Collections.Generic;
+﻿using System.Collections.Generic;
+using System.Linq;
+using UserService.CustomException;
 
 namespace UserService.Model
 {
     public class PatientAccount : UserAccount
     {
-        private DateTime DateOfRegistration { get; }
         private bool IsActivated { get; set; }
         private bool IsBlocked { get; set; }
         private string ImageName { get; }
         private IEnumerable<MaliciousAction> MaliciousActions { get; }
         
         public void Activate()
         {
-            throw new NotImplementedException();
+            if (IsBlocked) throw new ValidationException("Activation isn't possible because the patient is blocked.");
+            if (IsActivated) throw new ValidationException("Patient account is already activated.");
+            
+            IsActivated = true;
         }
 
         public void Block()
         {
-            throw new NotImplementedException();
+            if (!IsActivated) throw new ValidationException("Blocking unactivated patient account isn't possible.");
+            if (IsBlocked) throw new ValidationException("Patient account is already blocked.");
+
+            IsBlocked = true;
```

Comment:
Should you check if the patient is malicious here as well? Can patients that aren't malicious be blocked?

Reply:
You're right, I forgot that condition.